cpp/include/cudf/functions.h

```diff
@@ -949,3 +949,26 @@ gdf_error gdf_order_by(gdf_column** input_columns,
                        size_t       num_inputs,
                        gdf_column*  output_indices,
                        int          flag_nulls_are_smallest);
+
+/* --------------------------------------------------------------------------*
+ * @brief Finds the indices of the bins in which each value of the column
+ * belongs. For `x` in `col`, if `right == false` this function finds
+ * `i` such that `bins[i-1] <= x < bins[i]`. If `right == true`, it will find `i`
+ * such that `bins[i - 1] < x <= bins[i]`. Finally, if `x < bins[0]` or
+ * `x > bins[num_bins - 1]`, it sets the index to `0` or `num_bins`, respectively.
+ *
+ * @param[in] col gdf_column with the values to be binned
+ * @param[in] bins Device array of ascending bin boundaries
+ * @param[in] num_bins Length of bins
+ * @param[in] right Whether the intervals should include the left or right bin edge
+ * @param[out] out_col_indices Pre-allocated gdf_column to be filled with bin indices
+ *
+ * @return GDF_SUCCESS upon successful completion, otherwise an
+ *         appropriate error code.
+ *
+ * ----------------------------------------------------------------------------*/
+gdf_error gdf_digitize(gdf_column* col,
+                       void* bins,   // same type as col
+                       size_t num_bins,
+                       bool right,
+                       gdf_column* out_col_indices);
```

Comment:
Since there's no need for this to be type-erased, a `gdf_index_type[]` is more convenient.

Reply:
It looks like it's assumed that `out_col_indices` is a `uint32_t` column anyways. Therefore, this change makes even more sense.